hudi-flink/src/main/java/org/apache/hudi/operator/compact/CompactEvent.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.operator.compact;
+
+import org.apache.hudi.common.model.CompactionOperation;
+
+import java.io.Serializable;
+
+/**
+ * Represents a compact command from the compaction plan task {@link CompactionPlanOperator}.
+ */
+public class CompactEvent implements Serializable {
```

Comment:
Maybe `CompactionPlanEvent` make more sense?

Reply:
> Thanks @danny0405 for the awesome work. Hard to catch up on the review since you are making progress too fast :)
> Can't go into detail about this large PR too much until I get a chance to run this myself. Left some high-level comments.
> One concern is about the test cases. I feel like Flink writer is not as well tested as Spark, so the reliability is a bit concerning for me when we officially release this feature. Any plan to add more test cases?

Yes, we can add more test cases when more feature are introduced for Flink, such as `SQL connectors`, `INSERT OVERRIDE`, more kinds of key generators.